spec/models/ems_refresh/refreshers/ec2_refresher_security_group_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'spec_helper'
+
+describe ManageIQ::Providers::Amazon::CloudManager::Refresher do
+  let(:hashes) do
+    [
+      {:type                => 'ManageIQ::Providers::Amazon::CloudManager::SecurityGroup',
+       :ems_ref             => 'sg-e94055ac',
+       :name                => 'default',
+       :description         => 'default group',
+       :cloud_network       => nil,
+       :orchestration_stack => nil,
+       :firewall_rules      => [
+         {:direction             => nil,
+          :host_protocol         => nil,
+          :port                  => nil,
+          :end_port              => nil,
+          :source_security_group => nil
+         },
+         {:direction             => nil,
+          :host_protocol         => nil,
+          :port                  => nil,
+          :end_port              => nil,
+          :source_security_group => nil
+         },
+         {:direction             => 'inbound',
+          :host_protocol         => 'ICMP',
+          :port                  => -1,
+          :end_port              => -1,
+          :source_security_group => nil
+         }]
+      },
+      {:type                => 'ManageIQ::Providers::Amazon::CloudManager::SecurityGroup',
+       :ems_ref             => 'sg-2b87746f',
+       :name                => 'EmsRefreshSpec-SecurityGroup-OtherRegion',
+       :description         => 'EmsRefreshSpec-SecurityGroup-OtherRegion',
+       :cloud_network       => nil,
+       :orchestration_stack => nil,
+       :firewall_rules      => [
+         {:direction       => 'inbound',
+          :host_protocol   => 'TCP',
+          :port            => 0,
+          :end_port        => 65_535,
+          :source_ip_range => '0.0.0.0/0'
+         }]
+      }
+    ]
+  end
+
+  before(:each) do
+    _guid, _server, zone = EvmSpecHelper.create_guid_miq_server_zone
+    @ems = FactoryGirl.create(:ems_amazon, :zone => zone)
+  end
+
+  context '.save_security_groups_inventory' do
+    it 'should not raise an exception with empty firewall rules' do
+      expect do
+        EmsRefresh.save_security_groups_inventory(@ems, hashes)
+      end.to_not raise_error
```

Comment:
@Fryguy I updated the spec to check the output instead of confirming no exception is raised as you suggested.

Updating parser to avoid putting the empty firewall rules found in a security group into the hash seems like picking the fruit at the top of the tree to me. Building a test case would also be problematic. That said I'd be glad to do it if you feel it is worth the investment. Let me know and thank you again.